### ISSUE TITLE
craft a motherboard

### ISSUE THREAD

**Issue by moTechPlz** [MightyPirates/OpenComputers#944]
suggestion;

I really like how you can build your computer in this mod. How about the ability to craft your motherboard for the computer. Something like;
- Have different tier cases dictating what tier and size of the motherboard and amount of drive slots.
- Have different tier pcb's dictating the tier chips you can place on it. (Different sizes 2x2 or 3x3 maybe dictating the max amount of chips)
- Have different tier chips dictating the tier of cpu slot, memory slot, gpu slot etc and the tier of the drive slots in the case. (Having a tier 1 drive chip on the motherboard gives a case with for example 2 drive slots, 2 tier 1 drive slots. Maybe be able to insert a floppy drive in a tier 1, 2 or 3 slot)

So, craft a case, craft the pcb and chips, configure the motherboard and then place it in the case.
Ofcourse keeping the prebuild computer cases already in the game.

**Comment by Techokami** [comment]
Build a server then! Server blades are p.much motherboards.

**Comment by moTechPlz** [comment]
Yep, servers are cool. But you can't choose what tier one slot is for a give motherboard. Maybe make motherboards for servers too, 4x4.

**Comment by fnuecke** [comment]
So if I understand you correctly, you'd want configurable slots in cases, i.e. more fine-grained control over which slots are present in a case?

That seems like it over-complicates things a bit too much to me. I think the case tiers offer enough flexibility while keeping the things comparatively simple. From a rough estimation I also don't feel the time needed to implement this would be justified, given how strongly it overlaps with existing functionality. It'd also be difficult to avoid this being either unintuitive, assuming it would only apply to computer cases, or even more complex to implement assuming it should work in all devices.
Finally, the tier limits in the cases (and other devices) are quite deliberate. For example in cases they encourage using multiple computers that communicate with each other for more complex scenarios.

TL;DR: interesting idea, but too complicated and too hard to balance.
